lib/bolt/node/orch.rb

```diff
@@ -47,40 +47,29 @@ def _run_task(task, _input_method, arguments)
       state = node_result['state']
       result = node_result['result']
 
-      result_output = Bolt::Node::ResultOutput.new
-      result_output.stdout << result.to_json
       if state == 'finished'
-        Bolt::Node::Success.new(result.to_json, result_output)
+        exit_code = 0
       else
         # Try to extract the exit_code from _error
         begin
           exit_code = result['_error']['details']['exit_code'] || 'unknown'
         rescue NoMethodError
           exit_code = 'unknown'
         end
-        Bolt::Node::Failure.new(exit_code, result_output)
       end
+      Bolt::TaskResult.new(result.to_json, "", exit_code)
     end
 
     # run_task generates a result that makes sense for a generic task which
     # needs to be unwrapped to extract stdout/stderr/exitcode.
     #
     def unwrap_bolt_result(result)
-      task_result = JSON.parse(result.output.stdout.string)
-      if task_result['exit_code'].nil?
+      if result.error
         # something went wrong return the failure
         return result
       end
 
-      # Otherwise create a new result with the captured output
-      result_output = Bolt::Node::ResultOutput.new
-      result_output.stdout << task_result['stdout']
-      result_output.stderr << task_result['stderr']
-      if (task_result['exit_code']).zero?
-        Bolt::Node::Success.new(task_result['stdout'], result_output)
-      else
-        Bolt::Node::Failure.new(task_result['exit_code'], result_output)
-      end
+      Bolt::CommandResult.new(result.value['stdout'], result.value['stderr'], result.value['exit_code'])
```

Comment:
We expect to use Bolt::CommandResult for scripts as well?

Reply:
Yeah I considered renaming it. Any suggestions?

Reply:
It seems ok, just surprised me initially.